product_list_price_from_pricelist/models/product_pricelist.py

```diff
@@ -0,0 +1,125 @@
+                    product.write({"list_price": new_price})
+        return True
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    @api.constrains("min_quantity")
+    def _check_min_quantity_to_update_prices(self):
+        Company = self.env["res.company"]
+        for item in self.filtered(lambda item: item.min_quantity != 0):
+            pricelist_find = Company.search_count(
+                [("base_pricelist_compute_price_id", "=", item.pricelist_id.id)]
+            )
+            if pricelist_find:
+                raise ValidationError(
+                    _(
+                        "The pricelist is used as base pricelist to compute product prices. "
+                        "Its should not have items with minimum quantity, "
+                        "please check the pricelist %(pricelist)s with item %(name)s."
+                    )
+                    % {
+                        "pricelist": item.pricelist_id.name,
+                        "name": item.name or item.id,
+                    }
+                )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_pricelist_items = super().create(vals_list)
+        test_condition = not config["test_enable"] or (
+            config["test_enable"]
+            and self.env.context.get("test_compute_list_price_from_pricelist")
+        )
+        # If any field from the expected ones is changed,
+        # recompute the product's sales price.
+        if test_condition:
+            for pricelist in new_pricelist_items.pricelist_id:
+                used_in_company = bool(
+                    self.env["res.company"].search_count(
+                        [("base_pricelist_compute_price_id", "=", pricelist.id)]
+                    )
+                )
+                if used_in_company:
+                    pricelist._update_product_price_from_pricelist(new_pricelist_items)
+        return new_pricelist_items
+
+    def write(self, vals):
+        res = super().write(vals)
+        test_condition = not config["test_enable"] or (
+            config["test_enable"]
+            and self.env.context.get("test_compute_list_price_from_pricelist")
+        )
+        # If any field from the expected ones is changed,
+        # recompute the product's sales price.
+        if test_condition and set(vals.keys()).intersection(
+            self._get_fields_trigger_to_recompute_sale_price()
+        ):
+            for pricelist in self.pricelist_id:
+                used_in_company = bool(
+                    self.env["res.company"].search_count(
+                        [("base_pricelist_compute_price_id", "=", pricelist.id)]
+                    )
+                )
+                if used_in_company:
+                    pricelist._update_product_price_from_pricelist(self)
+        return res
+
+    @api.model
+    def _get_fields_trigger_to_recompute_sale_price(self):
```

Comment:
And please get used to add docstrings:

```suggestion
    def _get_fields_to_recompute_product_list_price(self):
        """Return the list of fields that will trigger the 
        recomputation of the products list price.
        """
```